src/errors.rs

```diff
@@ -31,6 +31,7 @@ pub enum Error {
 
 impl Error {
     /// Associtate the SQL request that caused the error
+    #[must_use]
```

Comment:
If I’m reading [the reference of must_use](https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute) correctly, that attribute could be added to `Error`, couldn’t it?

And then, all functions returning the struct would have to use the `Error`, I believe.

Reply:
> that attribute could be added to Error, couldn’t it?

In principle yes but `std::result::Result` is [already annotated with `#[must_use]](https://doc.rust-lang.org/std/result/#results-must-be-used), so I am not sure there is much to gain doing it again on the `Error` itself.

Reply:
Ok, makes sense.